src/paperoni/collection/filecoll.py

```diff
@@ -1,66 +1,47 @@
-import json
-from dataclasses import dataclass
-from functools import cached_property
+from dataclasses import dataclass, field
 from pathlib import Path
-from typing import Iterable
 
-from serieux import deserialize, serialize
+from serieux import deserialize, dump, load, serialize
 
-from ..model.classes import Paper
-from .tmpcoll import TmpCollection
-
-_id_types = {
-    "arxiv",
-    "dblp",
-    "doi",
-    "mlr",
-    "openalex",
-    "openreview",
-    "pmc",
-    "pubmed",
-    "semantic_scholar",
-    "uid",
-}
+from ..model.classes import CollectionPaper
+from .memcoll import MemCollection
 
 
 @dataclass
-class FileCollection(TmpCollection):
-    directory: Path
+class FileCollection(MemCollection):
+    directory: Path = field(compare=False)
+    _last_id: int = field(init=False, compare=False, default=None)
+    _papers: list[CollectionPaper] = field(init=False, default_factory=list)
+    _exclusions: set[str] = field(init=False, default_factory=set)
 
     def __post_init__(self):
-        if not self.directory.exists():
+        if self.directory is None:
+            # If no directory is provided, use the MemCollection implementation
+            self._file = None
+
+            super().__post_init__()
+            return
+
+        self._file = self.directory / "collection.yaml"
+
+        if self._file.exists():
+            self.__dict__.update(vars(load(type(self), self._file)))
+
+        else:
             self.directory.mkdir(exist_ok=True, parents=True)
-        self.papers_file = self.directory / "papers.json"
-        self.exclusions_file = self.directory / "exclusions.json"
-        if not self.papers_file.exists():
-            self.papers_file.write_text("[]")
-        self._papers = deserialize(list[Paper], self.papers_file)
-        if not self.exclusions_file.exists():
-            self.exclusions_file.write_text("[]")
-        self._exclusions = deserialize(set[str], self.exclusions_file)
+            self.commit()
 
-    @cached_property
-    def _by_title(self):
-        return {p.title: p for p in self._papers}
+    def commit(self) -> None:
+        if self._file is None:
+            super().commit()
+            return
 
-    @cached_property
-    def _by_link(self):
-        return {link: p for p in self._papers for link in p.links}
+        dump(type(self), self, dest=self._file)
 
-    def add_papers(self, papers: Iterable[Paper]) -> None:
-        papers = list(papers)
-        super().add_papers(papers)
-        if papers:
-            json.dump(
-                fp=open(self.papers_file, "w"),
-                obj=serialize(list[Paper], self._papers),
-            )
+    @classmethod
+    def serieux_deserialize(cls, obj: dict, ctx, call_next):
+        return deserialize(MemCollection, obj)
```

Comment:
```suggestion
        return call_next(MemCollection, obj, ctx)
```

Your way will work too, but the standard way is to use call_next and pass the ctx, that way it will also work with custom deserializers and all context variants.

Alternatively, since MemCollection is a dataclss, you can define `serieux_model`, which will implement serialize/deserialize/schema for you:

```python
@classmethod
def serieux_model(cls, call_next):
    return call_next(MemCollection)
```